File: Data Structures & Algorithms/design-hashmap/submission-1.go
type MyHashMap struct {
    arr []int
}

func Constructor() MyHashMap {
    tmp := make([]int,1000001)
    for idx := range tmp{
        tmp[idx]=-1
    }
    return MyHashMap{
        arr : tmp,
    }
}

func (this *MyHashMap) Put(key int, value int) {
    this.arr[key]=value
}

func (this *MyHashMap) Get(key int) int {
    return this.arr[key]
}

func (this *MyHashMap) Remove(key int) {
    this.arr[key]=-1
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Put(key,value);
 * param_2 := obj.Get(key);
 * obj.Remove(key);
 */

File: Data Structures & Algorithms/online-stock-span/submission-1.cpp
class StockSpanner {
private:
    stack<pair<int,int>> st; // {idx, value}
    int count = 0;
public:
    StockSpanner() {
        // do nothing still win
    }
    
    int next(int price) {
        if(st.empty()){
            st.push({count, price});
            count++;
            return 1;
        }
        while(!st.empty() && st.top().second <= price){
            st.pop();
        }
        int prevIdx = st.empty() ? -1 : st.top().first;
        st.push({count, price});
        count++;

        return count-prevIdx-1;
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */